### ISSUE TITLE
CF 2017 day 1 crime.net changes cause crash

### ISSUE THREAD

**Issue by Teacyn** [LBLuffy/HoloHud#38]
Crash date: 10/19/2017 2:53:29 PM

**PAYDAY 2 Crash log**

    Application has crashed: C++ exception
    mods/Holo/Hooks/CrimeNetManager.lua:31: attempt to index a nil value
    SCRIPT STACK
    init() @mods/base/req/core/Hooks.lua:267
    new() core/lib/utils/coreclass.lua:35
    create() lib/managers/menu/menucomponentmanager.lua:2179
    set_active_components() lib/managers/menu/menucomponentmanager.lua:429
    set_active_components() @mods/base/lua/MenuComponentManager.lua:19
    core/lib/managers/menu/coremenulogic.lua:89
    _execute_action_queue() core/lib/managers/menu/coremenulogic.lua:54
    update() core/lib/managers/menu/coremenulogic.lua:64
    update() core/lib/managers/menu/coremenumanager.lua:155
    update() lib/managers/menumanagerpd2.lua:17
    update() lib/setups/setup.lua:772
    update() lib/setups/menusetup.lua:294
    update() @mods/base/lua/MenuSetup.lua:6
    core/lib/setups/coresetup.lua:537
    -------------------------------
    Callstack:
             payday2_win32_release  (???)     ???                                                 
             payday2_win32_release  (???)     zip_get_name                                        
             payday2_win32_release  (???)     zip_get_name                                        
    -------------------------------
    Current thread: Main
    -------------------------------
    System information:
    	Application version : 1.79.376.
    	CPU : AMD FX(tm)-6300 Six-Core Processor (3 cores); SSE; SSE2; SSE3; SSSE3; SSE4.1; SSE5
    	DirectX : 12.0 
    	GPU : NVIDIA GeForce GTX 950 / nvldumd.dll[22.21.13.8541]
    	Language : english
    	Memory :     12286MB 1008KB
    	OS : 6.2.9200 () 0x100-0x1 (64 bits)
    	Physics : threaded
    	Renderer : DX9 threaded
    	Sound : Realtek Semiconductor Corp. (Speakers (Realtek High Definition Audio))

**Lua information**

Lua Hook: *present*

BLT Hook data: *present*

**Lua BLT Log (+/- 1 minute)**

    02:53:00 PM Log:  (..\src\InitiateState.cpp:490) Incoming HTTP Request/Request
    02:53:00 PM Log:  (..\src\InitiateState.cpp:504) http://download.paydaymods.com/download/latest/ITR - 28832
    02:53:00 PM Log:  (..\src\http\http.cpp:144) Launching Async HTTP Thread
    02:53:01 PM Lua: [Downloads] Finished download of Iter (Iter)
    02:53:01 PM Lua: [Downloads] Saving to downloads...
    02:53:01 PM Lua: [Downloads] Extracting...
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/changelog.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/english.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/french.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/german.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/italian.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/russian.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/loc/spanish.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/copactionwalk.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/copbrain.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/coplogictravel.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/coremissionmanager.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/coresequencemanager.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/coreworlddefinition.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/missionscriptelement.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/navigationmanager.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/_alex_2.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/_mia_1.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/_mus.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/lua/_roberts.lua
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/menu/
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/menu/options.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/mod.txt
    02:53:01 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Iter/Iter/tdlq.png
    02:53:01 PM Lua: [Downloads] Verifying...
    02:53:01 PM Lua: [Downloads] Removing old installation...
    02:53:01 PM Lua: [Downloads] Complete!
    02:53:19 PM Log:  (..\src\InitiateState.cpp:490) Incoming HTTP Request/Request
    02:53:19 PM Log:  (..\src\InitiateState.cpp:504) http://download.paydaymods.com/download/latest/FS - 28415
    02:53:19 PM Log:  (..\src\http\http.cpp:144) Launching Async HTTP Thread
    02:53:19 PM Lua: [Downloads] Finished download of Full Speed Swarm (Full Speed Swarm)
    02:53:19 PM Lua: [Downloads] Saving to downloads...
    02:53:19 PM Lua: [Downloads] Extracting...
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/changelog.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/english.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/french.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/german.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/italian.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/russian.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/loc/spanish.txt
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/actionspooc.lua
    02:53:19 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/aiattentionobject.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/basenetworksession.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/blackmarketmanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/blt_delayed_calls.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/blt_keybinds_manager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/carrydata.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/characterattentionobject.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/copactionwalk.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/copbase.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/copbrain.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/copdamage.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogicarrest.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogicattack.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogicbase.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogicidle.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogicsniper.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coplogictravel.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/copmovement.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/corecode.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/corecontrollerwrapper.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/corecontrollerwrappersettings.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coreelementarea.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coreelementshape.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coreevent.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coremath.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coremissionmanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coreportalmanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coreshapemanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/coretable.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/crimespreemanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/criminalsmanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/elementareatrigger.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/enemymanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/groupaistatebase.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/groupaistatebesiege.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/hudmanager.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/huskcopbrain.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/huskplayerinventory.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/huskplayermovement.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/huskteamaimovement.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/interactionext.lua
    02:53:20 PM Log:  (..\src\util\compression.cpp:125) Extracting to mods\_temp\Full Speed Swarm/Full Speed Swarm/lua/menumanager.lua
    02:53:20 PM Log:  (..\src\util\compress

**Comment by bamq** [comment]
Deleting the lines
```	
if not no_servers then
	self._panel:child("filter_button"):set_color(Holo:GetColor("TextColors/Menu"))
end
```
in Holo\Hooks\CrimeNetManager.lua (lines 30-32) fixes the issue for now.  
Seems they removed the old filters button so it points to a nil value.

**Comment by Luffyyy** [comment]
3.0 beta should have it fixed, feel free to report any issues of 3.0.
Edit: make sure you have BeardLib version 2.6 installed, get it from github.
Edit-Edit: I'll release BeardLib 2.6 in mws also, since auto updates are bork for now, manually download it
